oap-server/oal-rt/src/main/java/org/apache/skywalking/oal/rt/parser/ConditionExpression.java

```diff
@@ -50,11 +51,17 @@ public void addValue(String value) {
         }
     }
 
+    public void isNumber() {
+        number = true;
+    }
+
     public void enterMultiConditionValue() {
         values = new LinkedList<>();
     }
 
     public void exitMultiConditionValue() {
-        value = "new Object[]{" + String.join(",", values) + "}";
+        value = number ?
+            "new double[]{" + String.join(",", values) + "}" :
```

Comment:
Do you need `12D`/`12.11D`? When the number is very big.

Reply:
Done, currently only support digit.